File: web/src/utils/format.js
/** 格式化字节数 */
export function formatBytes(bytes) {
  if (bytes === null || bytes === undefined) return '-'
  if (bytes === 0) return '0 B'
  const units = ['B', 'KB', 'MB', 'GB', 'TB']
  const i = Math.min(Math.floor(Math.log(bytes) / Math.log(1024)), units.length - 1)
  const value = bytes / Math.pow(1024, i)
  return `${value >= 100 ? Math.round(value) : value.toFixed(1)} ${units[i]}`
}

/** 数字千分位 */
export function formatNumber(n) {
  if (n === null || n === undefined) return '-'
  return Number(n).toLocaleString('zh-CN')
}

/** 状态 → tag 类型 */
export const STATUS_TAG_TYPE = {
  DONE: 'success',
  RUNNING: 'primary',
  FAILED: 'danger',
  CANCELED: 'info',
  INTERRUPTED: 'warning',
  PENDING: 'info'
}

/** 状态 → 中文文案 */
export const STATUS_TEXT = {
  DONE: '完成',
  RUNNING: '运行中',
  FAILED: '失败',
  CANCELED: '已取消',
  INTERRUPTED: '已中断',
  PENDING: '等待中'
}

export function statusTagType(status) {
  return STATUS_TAG_TYPE[status] || 'info'
}

export function statusText(status) {
  return STATUS_TEXT[status] || status || '-'
}

/** 日期时间格式化 */
export function formatDateTime(v) {
  if (!v) return '-'
  const d = new Date(v)
  if (Number.isNaN(d.getTime())) return String(v)
  const pad = (x) => String(x).padStart(2, '0')
  return `${d.getFullYear()}-${pad(d.getMonth() + 1)}-${pad(d.getDate())} ${pad(d.getHours())}:${pad(d.getMinutes())}:${pad(d.getSeconds())}`
}

/** 耗时(毫秒 → 可读文案) */
export function formatDuration(startedAt, finishedAt) {
  if (!startedAt) return '-'
  const start = new Date(startedAt).getTime()
  const end = finishedAt ? new Date(finishedAt).getTime() : Date.now()
  if (Number.isNaN(start) || Number.isNaN(end)) return '-'
  let ms = Math.max(0, end - start)
  const s = Math.floor(ms / 1000)
  if (s < 60) return `${s}秒`
  const m = Math.floor(s / 60)
  if (m < 60) return `${m}分${s % 60}秒`
  const h = Math.floor(m / 60)
  return `${h}时${m % 60}分`
}

File: desktop/src/main/kotlin/com/example/dq/ui/components/Format.kt
package com.example.dq.ui.components

import com.example.dq.model.ScanStatus
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import kotlin.math.floor
import kotlin.math.ln
import kotlin.math.min
import kotlin.math.pow

/** 格式化字节数(平移自 web/src/utils/format.js) */
fun formatBytes(bytes: Long?): String {
    if (bytes == null) return "-"
    if (bytes == 0L) return "0 B"
    val units = arrayOf("B", "KB", "MB", "GB", "TB")
    val i = min(floor(ln(bytes.toDouble()) / ln(1024.0)).toInt(), units.size - 1)
    val value = bytes / 1024.0.pow(i)
    return if (value >= 100) "${Math.round(value)} ${units[i]}" else "${(value * 10).toLong() / 10.0} ${units[i]}"
}

/** 数字千分位 */
fun formatNumber(n: Long?): String =
    if (n == null) "-" else "%,d".format(n)

/** 状态 → 中文文案 */
fun statusText(status: ScanStatus?): String = when (status) {
    ScanStatus.DONE -> "完成"
    ScanStatus.RUNNING -> "运行中"
    ScanStatus.FAILED -> "失败"
    ScanStatus.CANCELED -> "已取消"
    ScanStatus.INTERRUPTED -> "已中断"
    ScanStatus.PENDING -> "等待中"
    null -> "-"
}

private val DATE_TIME_FMT: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")

/** 日期时间格式化 */
fun formatDateTime(v: LocalDateTime?): String = v?.format(DATE_TIME_FMT) ?: "-"

/** 耗时(起止时间 → 可读文案) */
fun formatDuration(startedAt: LocalDateTime?, finishedAt: LocalDateTime?): String {
    if (startedAt == null) return "-"
    val end = finishedAt ?: LocalDateTime.now()
    var s = maxOf(0L, java.time.Duration.between(startedAt, end).seconds)
    if (s < 60) return "${s}秒"
    val m = s / 60; s %= 60
    if (m < 60) return "${m}分${s}秒"
    return "${m / 60}时${m % 60}分"
}
